src/Microsoft.AspNet.Mvc.Localization/HtmlLocalizer.cs

```diff
@@ -0,0 +1,213 @@
+        /// <returns>The string with encoded arguments.</returns>
+        protected virtual string EncodeArguments([NotNull] string resourceString, [NotNull] object[] arguments)
+        {
+            var position = 0;
+            var length = resourceString.Length;
+            char currentCharacter;
+            StringBuilder tokenBuffer = null;
+            var outputBuffer = new StringBuilder();
+            var isToken = false;
+
+            while (position < length)
+            {
+                currentCharacter = resourceString[position];
+
+                position++;
+                if (currentCharacter == '}')
+                {
+                    if (tokenBuffer == null && !isToken)
+                    {
+                        tokenBuffer = new StringBuilder();
+                        isToken = true;
+                    }
+                    tokenBuffer.Append(currentCharacter);
+
+                    // Treat as escape character for }}
+                    while (position < length && resourceString[position] == '}')
+                    {
+                        tokenBuffer.Append(resourceString[position]);
+                        position++;
+                    }
+
+                    if (tokenBuffer != null && tokenBuffer.Length > 0)
+                    {
+                        var formattedString = string.Format(tokenBuffer.ToString(), arguments);
+                        if (formattedString.StartsWith("{") || formattedString.StartsWith("}"))
+                        {
+                            outputBuffer.Append(formattedString);
+                        }
+                        else
+                        {
+                            outputBuffer.Append(EncodeArgument(formattedString));
+                        }
+                    }
+
+                    if (position == length)
+                    {
+                        break;
+                    }
+                    else
+                    {
+                        currentCharacter = resourceString[position];
+                        position++;
+                        isToken = false;
+                        tokenBuffer = null;
+                    }
+                }
+
+                if (currentCharacter == '{')
+                {
+                    tokenBuffer = new StringBuilder();
+                    tokenBuffer.Append(currentCharacter);
+                    isToken = true;
+
+                    // Treat as escape character for {{
+                    while (position < length && resourceString[position] == '{')
+                    {
+                        tokenBuffer.Append(resourceString[position]);
+                        position++;
+                    }
+
+                    if (resourceString[position] < '0' || resourceString[position] > '9')
+                    {
+                        if (tokenBuffer != null && tokenBuffer.Length > 0)
```

Comment:
Remove condition. `tokenBuffer` cannot be `null` here. And `tokenBuffer` always contains at least one character.